openmrs/src/test/java/org/motechproject/openmrs/tasks/OpenMRSTaskDataProviderTest.java

```diff
@@ -265,72 +279,155 @@ public void shouldReturnNullWhenWrongLookupNameForProvider() {
+
+        Object object = taskDataProvider.lookup(className + '-' + CONFIG_NAME, BY_UUID_AMD_PROGRAM_NAME, lookupFields);
+
+        verify(programEnrollmentService).getProgramEnrollmentByPatientUuid(eq(CONFIG_NAME), eq(DEFAULT_UUID));
 
         assertEquals(expected.get(0), object);
     }
 
+    @Test
+    public void shouldReturnProgramEnrollmentForPatientMotechIdAndProgramName() {
+        String className = ProgramEnrollment.class.getSimpleName();
+
+        Map<String, String> lookupFields = new HashMap<>();
+        lookupFields.put(PATIENT_MOTECH_ID, DEFAULT_MOTECH_ID);
+        lookupFields.put(PROGRAM_NAME, PROGRAM_DEFAULT_NAME);
+
+        List<ProgramEnrollment> expected = prepareProgramEnrollments();
+        when(programEnrollmentService.getProgramEnrollmentByPatientMotechId(eq(CONFIG_NAME), eq(DEFAULT_MOTECH_ID)))
+                .thenReturn(expected);
+
+        Object object = taskDataProvider.lookup(className + '-' + CONFIG_NAME, BY_MOTECH_ID_AND_PROGRAM_NAME, lookupFields);
+
+        verify(programEnrollmentService).getProgramEnrollmentByPatientMotechId(eq(CONFIG_NAME), eq(DEFAULT_MOTECH_ID));
+
+        assertEquals(expected.get(0), object);
+    }
+
+    @Test
+    public void shouldReturnNotEnrolledWhenProgramEnrollmentNotFoundForLookupByUuidAndProgramName() {
+        String className = ProgramEnrollment.class.getSimpleName();
+
+        Map<String, String> lookupFields = new HashMap<>();
+        lookupFields.put(PATIENT_UUID, DEFAULT_UUID);
+        lookupFields.put(PROGRAM_NAME, PROGRAM_ANOTHER_NAME);
+
+        when(programEnrollmentService.getProgramEnrollmentByPatientUuid(eq(CONFIG_NAME), eq(DEFAULT_UUID)))
+                .thenReturn(prepareProgramEnrollments());
+
+        Object object = taskDataProvider.lookup(className + '-' + CONFIG_NAME, BY_UUID_AMD_PROGRAM_NAME, lookupFields);
+
+        verify(programEnrollmentService).getProgramEnrollmentByPatientUuid(eq(CONFIG_NAME), eq(DEFAULT_UUID));
+
+        assertEquals(ProgramEnrollment.NOT_ENROLLED, ((ProgramEnrollment) object).getEnrolled());
+    }
+
+    @Test
+    public void shouldReturnNotEnrolledWhenProgramEnrollmentNotFoundForLookupByMotechIdAndProgramName() {
+        String className = ProgramEnrollment.class.getSimpleName();
+
+        Map<String, String> lookupFields = new HashMap<>();
+        lookupFields.put(PATIENT_MOTECH_ID, DEFAULT_MOTECH_ID);
+        lookupFields.put(PROGRAM_NAME, PROGRAM_ANOTHER_NAME);
+
+        when(programEnrollmentService.getProgramEnrollmentByPatientMotechId(eq(CONFIG_NAME), eq(DEFAULT_MOTECH_ID)))
+                .thenReturn(prepareProgramEnrollments());
+
+        Object object = taskDataProvider.lookup(className + '-' + CONFIG_NAME, BY_MOTECH_ID_AND_PROGRAM_NAME, lookupFields);
+
+        verify(programEnrollmentService).getProgramEnrollmentByPatientMotechId(eq(CONFIG_NAME), eq(DEFAULT_MOTECH_ID));
+
+        assertEquals(ProgramEnrollment.NOT_ENROLLED, ((ProgramEnrollment) object).getEnrolled());
```

Comment:
I think the best way is to create two methods in ProgramEnrollment :
isEnrolled() and setEnrolled(boolean) and in method logic you will be convert boolean to string or string to boolean